### PR TITLE
Self hosted Kubernetes Documentation

### PR DESCRIPTION
**Pull request by glaubinix**
For now, letting me know anything that is missing would be super helpful!

### CODE REVIEW

File: docs/self-hosted/kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  * At least 40GB disk space
+  * Ports 443 and 8800 must be accessible
+  * Must be reachable at the chosen domain names from your local machine
+
+<!-- See https://kurl.sh/docs/install-with-kurl/system-requirements#networking-requirements and https://docs.replicated.com/enterprise/installing-general-requirements -->
+* If your firewall restricts external connections the following domains must be accessible from the server:
+  * amazonaws.com
+  * k8s.gcr.io
+  * k8s.kurl.sh
+  * hub.docker.com/
```
Comment (pscheit) [inline docs/self-hosted/kubernetes.md:39]:
```suggestion
  * hub.docker.com
```

File: docs/self-hosted/kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+### Installation
+
+The easiest way to install Replicated is the easy install script. You can run it using the following command:
+
+```
+curl -sSL https://kurl.sh/privatepackagistkots | sudo bash
+```
+
+To learn more about options for the easy install script, please refer to the Replicated manual on Installing Replicated at https://help.replicated.com/docs/kubernetes/customer-installations/installing/.
```
Comment (pscheit) [inline docs/self-hosted/kubernetes.md:51]:
```suggestion
To learn more about options for the easy install script, please refer to the Replicated manual on Installing Replicated at [https://help.replicated.com/docs/kubernetes/customer-installations/installing/](https://help.replicated.com/docs/kubernetes/customer-installations/installing/).
```

File: docs/self-hosted/kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+## Installing Private Packagist Self-Hosted on an existing Kubernetes cluster
+
+### Requirements
+
+* A Kubernetes v1.23.x cluster
+
+<!-- See https://docs.replicated.com/enterprise/installing-general-requirements -->
+* If your firewall restricts external connections the following domains must be accessible from the server:
+  * hub.docker.com/
```
Comment (pscheit) [inline docs/self-hosted/kubernetes.md:63]:
```suggestion
  * hub.docker.com
```

**Review by pscheit — CHANGES_REQUESTED** [review]

**Comment by zanbaldwin** [comment]
- Would probably not make port 8800 a requirement - mention that in order to access the admin panel remotely then it would need to be forwarded to a port of their choosing (might be nice in the future to have a handy `kubectl apply` YAML file to provide instead of having to set up Nginx configuration).
- If the storage class used is Longhorn, at this point in time it does not support Kubernetes v..25+
- The Kots admin panel is available on install, but stops after a Ctrl-C. Is there a way for it to run continuously in the background like it does on regular Replicated?

**Comment by glaubinix** [comment]
Thanks, will update the port thing. It's still required for embedded cluster. The storage class is up to them, so nothing we have to document. Unsure about the admin panel. Will look into this but I think the idea might be to just start it while you need it and shut it down when not because you don't want to expose services if you don't need them.

**Review by pscheit — APPROVED** [review]
historic day =)

**Pull request merged**